metrics/dbmetrics.go

```diff
@@ -102,3 +116,50 @@ func (e *dbCallMetrics) elapsed() time.Duration {
 	}
 	return e.endTime.Sub(e.startTime)
 }
+
+// NewPostgreDBMetrics returns new PostgreDBMetrics.
+func NewPostgreDBMetrics(metricsProvider Provider, dbName string) *PostgreDBMetrics {
+	l := []string{dbMetricLabelInstance}
+	MaxOpenConnections := metricsProvider.NewHistogram("postgres_db_stat_max_open_connections",
+		fmt.Sprintf("Maximum open connections on %s", dbName), prometheus.DefBuckets, l...)
+	OpenConnections := metricsProvider.NewHistogram("postgres_db_stat_open_connections",
+		fmt.Sprintf("Established connections both in use and idle on %s", dbName), prometheus.DefBuckets, l...)
+	InUse := metricsProvider.NewHistogram("postgres_db_stat_in_use",
+		fmt.Sprintf("Connections currently in use on %s", dbName), prometheus.DefBuckets, l...)
+	Idle := metricsProvider.NewHistogram("postgres_db_stat_idle",
+		fmt.Sprintf("Idle connections on %s", dbName), prometheus.DefBuckets, l...)
+	WaitCount := metricsProvider.NewHistogram("postgres_db_stat_wait_count",
+		fmt.Sprintf("Total connections waited for on %s", dbName), prometheus.DefBuckets, l...)
+	WaitDuration := metricsProvider.NewHistogram("postgres_db_stat_wait_duration",
+		fmt.Sprintf("Total time blocked waiting for a new connection on %s", dbName), prometheus.DefBuckets, l...)
+	MaxIdleClosed := metricsProvider.NewHistogram("postgres_db_stat_max_idle_closed",
+		fmt.Sprintf("Total connections closed due to SetMaxIdleConns on %s", dbName), prometheus.DefBuckets, l...)
+	MaxIdleTimeClosed := metricsProvider.NewHistogram("postgres_db_stat_max_idle_time_closed",
+		fmt.Sprintf("Total connections closed due to SetConnMaxIdleTime on %s", dbName), prometheus.DefBuckets, l...)
+	MaxLifetimeClosed := metricsProvider.NewHistogram("postgres_db_stat_max_lifetime_closed",
+		fmt.Sprintf("Total connections closed due to SetConnMaxLifetime on %s", dbName), prometheus.DefBuckets, l...)
+	return &PostgreDBMetrics{
+		MaxOpenConnections: MaxOpenConnections,
+		OpenConnections:    OpenConnections,
+		InUse:              InUse,
+		Idle:               Idle,
+		WaitCount:          WaitCount,
+		WaitDuration:       WaitDuration,
+		MaxIdleClosed:      MaxIdleClosed,
+		MaxIdleTimeClosed:  MaxIdleTimeClosed,
+		MaxLifetimeClosed:  MaxLifetimeClosed,
+	}
+}
+
+func (dbMetric *PostgreDBMetrics) ObservePostgreDBMetric(dbType string, db *sql.DB) {
+	dbStats := db.Stats()
+	dbMetric.MaxOpenConnections.With(map[string]string{dbMetricLabelInstance: dbType}).Observe(float64(dbStats.MaxOpenConnections))
```

Comment:
Seems we repeat `map[string]string{dbMetricLabelInstance: dbType}` this in every lines, can we put it into local var instead? Just to make the line shorter.

Reply:
code updated